### ISSUE TITLE
create export file for qualititive expression data

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#1212]
create export file for qualitative expression data

**Comment by kimrutherford** [comment]
Are these columns OK?  That's what we require for the input files:
 - Systematic ID
 - Gene name
 - Type
 - Evidence
 - Qualifier
 - Extension
 - Reference 
 - Taxon
 - Date 

Maybe we can rename "Type" as it's a bit generic.  Maybe "Protein or RNA" would be clearer?

**Comment by ValWood** [comment]
Looks good!

**Comment by kimrutherford** [comment]
Here's an example of what the file will look like:
https://curation.pombase.org/kmr44/qualitative_gene_expression.tsv

> And there's one term that has only one annotation:
> "alternative transcripts, alternative mRNA processing"

That annotation isn't being write to the file at the moment because it doesn't have a "qualifier" value like "increased".  I can include it in the output with a blank qualifier if that makes sense.

**Comment by kimrutherford** [comment]
I've added this qualitative data file to the release directory:
https://www.pombase.org/monthly_releases/pombase-2025-02-01/gene_expression/

(I'll update the file contents there if changes are needed)

And I've changed the README to mention the file.

**Comment by ValWood** [comment]
and ribosomal density should probably be "protein" as the qualifiers are

for eg re PomGeneEx:0000027 ribosomal density decreased

**Comment by ValWood** [comment]
And there's one term that has only one annotation:
"alternative transcripts, alternative mRNA processing"
https://www.pombase.org/term/PBO:0108543
That's annotated in chromosome1.contig SPAC22A12.15c

 needs fixing.
it  had the wrong prefix in Artemis, so it got exported as an expression annotation.
It should have been.

"alternative transcripts, alternative mRNA processing"
I will delete the row and add it back in Atremis

**Comment by kimrutherford** [comment]
> I'm not sure that "transcription" should be in that column

In [the documentation](https://www.pombase.org/documentation/qualitative-gene-expression-data-bulk-upload-format) we give the examples of "protein" or "RNA" in the Type column.  What should we put in the type column of the export file if the term is "transcription decreased"?  "RNA"?

> and ribosomal density should probably be "protein" as the qualifiers are
> for eg re PomGeneEx:0000027 ribosomal density decreased

So for that case "protein" in the type column.  What should go in the qualifier column?  "ribosomal density decreased"?

**Comment by ValWood** [comment]
>In [the documentation](https://www.pombase.org/documentation/qualitative-gene-expression-data-bulk-upload-format) we give the examples of "protein" or "RNA" in the Type column. What should we put in the type column of the export file if the term is "transcription decreased"? "RNA"?

yes to this

**Comment by ValWood** [comment]
> and ribosomal density should probably be "protein" as the qualifiers are
for eg re PomGeneEx:0000027 ribosomal density decreased
So for that case "protein" in the type column. What should go in the qualifier column? "ribosomal density decreased"?


I think for this one it is OK to have type protein even though we are talking about ribosome density , because that will be clear from the qualifier.
@PCarme  what do you think? Would this be confusing? Do we need a type "ribosome density"?

**Comment by ValWood** [comment]
Re
"alternative transcripts, alternative mRNA processing"
I will delete the row and add it back in Atremis

where is the file? I can't see the qualitative data from the contigs in supporting_data or extenal_data

**Comment by PCarme** [comment]
> > and ribosomal density should probably be "protein" as the qualifiers are
> > for eg re PomGeneEx:0000027 ribosomal density decreased
> > So for that case "protein" in the type column. What should go in the qualifier column? "ribosomal density decreased"?
> 
> I think for this one it is OK to have type protein even though we are talking about ribosome density , because that will be clear from the qualifier. [@PCarme](https://github.com/PCarme) what do you think? Would this be confusing? Do we need a type "ribosome density"?

I agree, what was measured will be clear from the qualifier, and it is used to reflect a change in protein level anyway

**Comment by ValWood** [comment]
> Re
"alternative transcripts, alternative mRNA processing"
I will delete the row and add it back in Atremis

I am not going to add this back to artemis. I can't find any evidence of alternative transcripts giving rise to alternative products, so we  would not curate this. So Kim could you just  delete the row from the expression file.

tks
v

**Comment by kimrutherford** [comment]
> I am not going to add this back to artemis. I can't find any evidence of alternative transcripts giving rise to alternative products, so we would not curate this. So Kim could you just delete the row from the expression file.

It's in `chromosome1.contig`.  Should I delete it from there?  It will disable from the expression export file once it's gone from the contig file.

```
FT                   /controlled_curation="term=gene expression, alternative
FT                   transcripts, alternative mRNA processing;
FT                   db_xref=PMID:23066505; date=20170215"
```

**Comment by ValWood** [comment]
doing it now. so the expression infor is still in the contics. Got it!

**Comment by kimrutherford** [comment]
The expression info curated by you is still in the contigs.
There is a bunch of other annotation in `pombe-embl/external_data/Quantitative_gene_expression_data/` and  `pombe-embl/external_data/qualitative_gene_expression_data/`

**Comment by kimrutherford** [comment]
>> and ribosomal density should probably be "protein" as the qualifiers are
>> for eg re PomGeneEx:0000027 ribosomal density decreased
>> So for that case "protein" in the type column. What should go in the qualifier column? "ribosomal density decreased"?
>
> I think for this one it is OK to have type protein even though we are talking about ribosome density , because that will be clear from the qualifier.

What should go in the qualifier column in these cases?

**Comment by ValWood** [comment]
don't they already have a qualifier?

**Comment by kimrutherford** [comment]
There isn't a separate qualifier, just a term like:
```
PomGeneEx:0000027 ribosomal density decreased
```
or
```
PomGeneEx:0000018 protein level increased
```

For terms like "ribosomal density decreased" should the type be "protein" and qualifier be "ribosomal density decreased"? 

I understand the easy cases like `PomGeneEx:0000022 protein absent` - the type is "protein" and the qualifier is "absent" but I didn't know what to do in the "ribosomal density" and "transcription" cases.

The full list of terms is here:
 - https://github.com/pombase/pombase-chado/issues/1212#issuecomment-2641904718

**Comment by kimrutherford** [comment]
The export contains all the qualitative gene expression annotation from all the datasets.  Here's what it currently looks like:
https://www.pombase.org/monthly_releases/2025/pombase-2025-02-01/gene_expression/qualitative_gene_expression.tsv

**Comment by ValWood** [comment]
OK how it is here fine
https://www.pombase.org/reference/PMID:33313903

the protein/RNA label is just to determine which section of the gene page it appears in

SO, I think all that needs to change is the 'type' and the qualifiers should be correct?

**Comment by kimrutherford** [comment]
> OK how it is here fine
> https://www.pombase.org/reference/PMID:33313903

I don't understand.  I didn't think there was a problem with the website.  I'm just trying to work out what values to put in each column of the export file.

> SO, I think all that needs to change is the 'type'

Sorry, I still don't understand.  Which type needs to change and what should it change to?  Does the annotation need to change?

>  and the qualifiers should be correct?

My question is what should the qualifiers be?  What should go in the qualifier column?:

>>> and ribosomal density should probably be "protein" as the qualifiers are
>>> for eg re PomGeneEx:0000027 ribosomal density decreased
>>> So for that case "protein" in the type column. What should go in the qualifier column? "ribosomal density decreased"?
>>
>> I think for this one it is OK to have type protein even though we are talking about ribosome density , because that will be clear from the qualifier.
>
> What should go in the qualifier column in these cases?

I need help with just these 5 terms:

|type                                                              |
|------------------------------------------------------------------|
|PomGeneEx:0000027 ribosomal density decreased                     |
|PomGeneEx:0000026 ribosomal density increased                     |
|PBO:0091902 transcription unchanged                               |
|PBO:0091483 transcription decreased                               |
|PBO:0091437 transcription increased                               |

I'm off to bed now.  Maybe it will make sense in the morning.

**Comment by kimrutherford** [comment]
Yep, I think we need a Zoom chat.  I'm around any time on Thursday morning.

**Comment by kimrutherford** [comment]
> PBO:0091902 transcription unchanged

I've changed "term=gene expression, transcription ..." to "term=gene expression, RNA level ..." in the contig files.  13 occurrences.

Fixed the documentation to mention "ribosomal density".

And changed the qualitative gene expression writer to standardise on "RNA" and "protein".

It should be OK in the morning.  I'll check then.

**Comment by kimrutherford** [comment]
I think this is all done.
